tests/DCPS/GroupPresentation/Writer.cpp

```diff
@@ -143,7 +143,7 @@ Writer::svc()
   }
 
   finished_instances_ ++;
-
+  ACE_OS::sleep (1);
```

Comment:
Is this still required? I think several of our other tests wait until subscriber match count is zero, which might be a better fit for the intended behavior here... "wait until the subscriber has gotten all our updates"

Reply:
This is a deeper sleep. This is in the service object's thread function, and is giving the subscriber a chance to handle the end_coherence message before discarding  the association with the writer. In the subscriber app, there are multiple threads handling I/O events at the CORBA level. In the DataReaderImpl, there are two functions that serve as entry points from the infrastructure that come into conflict and motivated me to add the sleep. First is data_received() which serves as a message dispatcher. It tries very early on to inform the system that a given writer is still active. The second function that is accessed by the infrastructure is remove_associations(). The problem is that there are two infrastructure threads, and without the sleep, sometimes a thread obtains the end_coherency message but is swapped out by the OS before it can enter data_received(). Meanwhile the second thread detects a connection closure or whatever happens at the end of the writer's service function and it calls remove_associations() and cleans up the writer info before the first thread can process its message. When that first thread does get a chance to run, it has no way to dispatch the message as the necessary associations have been cleaned up. This sleep makes it so the end_coherency message is always processed before the associations are purged.

Reply:
I have removed this sleep. The need for it no longer exists on the subscriber side.